### PR TITLE
feat: HTTP mux wiring — A2A, MCP, Agent Card (#54)

### LINKED ISSUE #54: A2A interaction lifecycle: autonomous triage → RCA presentation → user-driven next step
## Summary

Define the end-to-end behavioral model for how the API Frontend operates as an A2A agent during an investigation request. AF works autonomously through triage and RR creation, streams the CRD lifecycle back to the user, then **yields control** when the RCA is presented — waiting for the user to decide the next step.

This is the Cursor/Claude Code interaction pattern: autonomous work → present results → wait for the human.

## Interaction Model

```
User: "investigate problems in namespace payments"
  │
  │  A2A task status: "working"
  ▼
┌─────────────────────────────────────────────────┐
│ Phase 1: Triage (AF autonomous, LLM-driven)     │
│                                                  │
│  AF LLM → get_events(ns=payments)               │
│  AF LLM → list_pods(ns=payments)                │
│  AF LLM → get_deployments(ns=payments)          │
│  AF LLM correlates → identifies affected         │
│           resources                              │
│  AF LLM → create_rr(deploy/payment-svc, ...)    │
│                                                  │
│  SSE: "RR-4821 created for deploy/payment-svc"  │
└─────────────────────────────────────────────────┘
  │
  │  A2A task status: "working"
  ▼
┌─────────────────────────────────────────────────┐
│ Phase 2: CRD Lifecycle Watch (AF passive)        │
│                                                  │
│  AF watches RR-4821 via kubernaut_watch          │
│                                                  │
│  SSE: "SP: classified OOMKilled, severity high"  │
│  SSE: "AA: policy approved investigation"        │
│  SSE: "KA: investigating..."                     │
│  SSE: "KA: found memory limit 256Mi,             │
│         peak usage 312Mi, OOMKilled 12x/24h"    │
│  SSE: "KA: investigation complete"               │
│  SSE: "KA: recommended: increase-memory-limit"   │
└─────────────────────────────────────────────────┘
  │
  │  A2A task status: "input-required"  ← KEY MOMENT
  ▼
┌─────────────────────────────────────────────────┐
│ Phase 3: RCA Presentation + Yield                │
│                                                  │
│  AF presents RCA findings:                       │
│    Resource: deploy/payment-svc                  │
│    Root cause: memory limit 256Mi exceeded       │
│    OOMKilled 12x in 24h, peak usage 312Mi       │
│    Recommended workflow: increase-memory-limit    │
│                                                  │
│  AF STOPS. Waits for user instruction.           │
└─────────────────────────────────────────────────┘
  │
  │  User responds (A2A tasks/send with same task ID)
  ▼
┌─────────────────────────────────────────────────┐
│ Phase 4: User-Driven Next Step                   │
│                                                  │
│  Option A: "accept, proceed with the workflow"   │
│    → AF proxies kubernaut_select_workflow         │
│    → A2A task status: "completed"                │
│                                                  │
│  Option B: "investigate further — check 7d       │
│             Prometheus metrics"                  │
│    → AF proxies kubernaut_investigate             │
│      (action=start or action=message)            │
│    → Returns to Phase 2 (watch + stream)         │
│    → Yields again at next RCA/finding            │
│                                                  │
│  Option C: "cancel, this is a known issue"       │
│    → AF calls cancel_remediation                 │
│    → A2A task status: "completed"                │
└─────────────────────────────────────────────────┘
```

## A2A Task Status Mapping

| Phase | A2A task status | AF behavior |
|-------|----------------|-------------|
| Triage + RR creation | `working` | AF streams `task-status-update` events |
| CRD lifecycle watch | `working` | AF streams investigation progress via SSE |
| RCA presented | **`input-required`** | AF yields — task pauses, waits for user |
| User responds → workflow selected | `working` → `completed` | AF acts on user's choice |
| User responds → continue investigating | `working` → `input-required` | Loop back to watch + yield |
| User responds → cancel | `completed` | AF cancels remediation |

## AF System Prompt Guidance

The AF system prompt must enforce:

1. **Autonomous triage**: query K8s, identify problems, create RRs without asking
2. **Passive watch**: stream CRD lifecycle updates without intervening
3. **Yield at RCA**: when investigation completes, present findings and STOP
4. **Never auto-select workflows**: always wait for explicit user instruction
5. **Never auto-approve remediations**: approval gates are non-negotiable (per kubernaut#818 design principles)
6. **Support iterative investigation**: if user wants to dig deeper, proxy to KA's interactive mode and yield again at the next finding

## Tools Involved Per Phase

| Phase | AF tools | Mode |
|-------|----------|------|
| Triage | `get_events`, `list_pods`, `get_deployments` | LLM-driven (AF's own tools) |
| Create RR | `create_rr` | LLM-driven (AF's own tool, direct K8s API) |
| Watch lifecycle | `kubernaut_watch` | Passive stream (AF's native tool) |
| Present RCA | (LLM synthesis) | LLM-driven |
| Select workflow | `kubernaut_select_workflow` | Proxy to KA |
| Continue investigating | `kubernaut_investigate` | Proxy to KA |
| Cancel | `cancel_remediation` | AF's own tool, direct K8s API |

## Dependencies

- #52 — NL signal intake (Phase 1: triage + RR creation)
- #13 — A2A inbound endpoint (protocol layer that carries this interaction)
- #3 — MCP tool surface (tools used across all phases)
- #19 — Native K8s CRD tools (create_rr, cancel, list, get)
- #7 — Session lifecycle alignment (KA session mapping for Phase 4 interactive mode)

## Related

- jordigilh/kubernaut#818 — Vision: safety architecture principles (no auto-approval)
- #14 — API Frontend service boundary
- #15 — A2A protocol support
- #16 — Kubernaut Console (UI consumer of this interaction model)

### PR DESCRIPTION
**Pull request by jordigilh**
## Summary

- **HTTP Router** (`internal/handler`): Structured mux with public endpoints (`/healthz`, `/readyz`, `/metrics`, `/.well-known/agent-card.json`) and authenticated endpoints (`/a2a/invoke`, `/mcp`). Metrics middleware records `af_http_requests_total` and `af_http_request_duration_seconds`.
- **A2A Launcher** (`internal/launcher`): Wraps ADK executor in the a2a-go JSON-RPC transport. `BeforeExecuteCallback` propagates user identity from auth middleware; `AfterExecuteCallback` provides structured error logging. `AutoCreateSession: true` delegates session lifecycle to the ADK.
- **MCP Server** (`internal/handler/mcp.go`): Streamable HTTP handler with 14 `kubernaut_*` tool stubs registered (real tool bridge deferred to PR6+).
- **Agent Card** (`internal/handler/agentcard.go`): Static JSON endpoint at `/.well-known/agent-card.json` declaring skills, auth requirements, and streaming capabilities per A2A spec.
- **main.go rewrite**: Replaces inline routing with `handler.NewRouter`, assembling root agent, session service, A2A handler, MCP handler, Agent Card, and auth middleware chain.

## Test plan

- [x] 46 Ginkgo specs pass with `-race` (10 router + 14 launcher + 12 MCP + 10 agent card)
- [x] Full project `go test ./... -race` passes
- [x] `golangci-lint run ./...` — 0 issues
- [x] `gosec -severity medium -confidence medium` — 0 findings
- [x] `go mod tidy` clean
- [x] `go build ./...` clean
- [x] IEEE 829 test plan authored at `docs/tests/54/test_plan.md`

## Commits

1. `docs:` Test plan (Phase 0)
2. `feat(handler):` Router + middleware + health + 10 specs
3. `feat(launcher):` A2A executor + model config + 14 specs
4. `feat(handler):` MCP server + 14 tool stubs + 12 specs
5. `feat(handler):` Agent Card + 10 specs
6. `refactor(main):` Wire all components into handler.NewRouter

Closes #54

Made with [Cursor](https://cursor.com)

### CODE REVIEW

**Review by jordigilh — COMMENTED** [review]
## 8-Persona PR Review — PR #80

| Persona | P1 | P2 | P3 |
|---|---|---|---|
| QE | 0 | 2 | 2 |
| Prod Security | 1 | 0 | 1 |
| Product | 0 | 0 | 2 |
| UX | 0 | 0 | 1 |
| Architecture | 0 | 1 | 1 |
| SRE/Ops | 0 | 1 | 1 |
| Release/Change Mgmt | 0 | 2 | 1 |
| FedRAMP | 0 | 1 | 1 |
| **Total** | **1** | **7** | **10** |

**Verdict: Request changes** — 1 P1 (unbounded Prometheus label cardinality via raw URL path) must be fixed before merge. The 7 P2s cover missing test coverage, a streaming-breaking `ResponseWriter` wrapper, audit gaps, and stub tools in production. All are actionable.

### P1 (must-fix)
- **SEC-1**: `metricsMiddleware` uses raw `r.URL.Path` as Prometheus label → unbounded cardinality DoS

### P2 Summary
- **QE-1**: `AllReady` exported function has no test
- **QE-2**: `RouterConfig.validate()` 6 error paths have no individual test
- **ARCH-2**: `statusRecorder` doesn't implement `http.Flusher` → will break MCP streaming
- **OPS-1**: `AfterExecuteCallback` error log lacks task ID, session ID, method
- **REL-1**: MCP exposes 14 stub tools unconditionally in production
- **REL-2**: No CHANGELOG or release notes for 5 new endpoints
- **FED-1**: No audit events for A2A task lifecycle or MCP tool invocations

### P3 Summary
- **QE-3**: `UT-AF-220-004` dead code — `createMCPServerWithTools()` result unused
- **QE-4**: `UT-AF-210-014` is a tautology (assign+read struct field)
- **SEC-2**: `buildBeforeExecuteCallback` discards `UserIdentityFromContext` result — dead code
- **PROD-1**: `/readyz` failure message changed from `"JWKS circuit breaker open"` to `"not ready"`
- **PROD-2**: `ModelConfig` + `DefaultModelConfig()` exported but never called from production code
- **UX-1**: Agent card `url` field hardcodes `https://localhost:{PORT}`
- **ARCH-1**: `AllReady` exported but never called from production code
- **OPS-2**: `/readyz` only checks JWKS, not A2A/MCP/session readiness
- **REL-3**: New env vars (`GCP_PROJECT`, `GCP_REGION`, `KA_BASE_URL`, etc.) undocumented
- **FED-2**: MCP tool stubs use empty input schemas — must have proper validation when bridge is wired

File: internal/handler/middleware.go [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+func metricsMiddleware(reg *metrics.Registry, next http.Handler) http.Handler {
+	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		start := time.Now()
+		rec := &statusRecorder{ResponseWriter: w, status: http.StatusOK}
+
+		next.ServeHTTP(rec, r)
+
+		duration := time.Since(start).Seconds()
+		status := strconv.Itoa(rec.status)
+		path := r.URL.Path
```
Comment (jordigilh) [inline internal/handler/middleware.go]: **[P1 SEC-1] Unbounded Prometheus label cardinality via raw `r.URL.Path`**

`path := r.URL.Path` is used directly as a Prometheus label. An attacker can send requests to arbitrary paths (`/random/1`, `/random/2`, ...) creating unlimited metric time series, exhausting Prometheus memory (cardinality bomb).

**Fix**: Normalize the path to a fixed set of known routes:
```go
func normalizePath(p string) string {
    switch {
    case p == "/healthz", p == "/readyz", p == "/metrics":
        return p
    case strings.HasPrefix(p, "/a2a/"):
        return "/a2a/invoke"
    case p == "/mcp":
        return "/mcp"
    case p == "/.well-known/agent-card.json":
        return "/.well-known/agent-card.json"
    default:
        return "unknown"
    }
}
```
Then use `normalizePath(r.URL.Path)` instead of the raw path.

File: internal/handler/health.go [hunk 2]
```diff
@@ -0,0 +1,35 @@
+package handler
+
+import "net/http"
+
+// ReadyChecker is a function that reports whether the service is ready to
+// accept traffic. Multiple checkers can be composed with AllReady.
+type ReadyChecker func() bool
+
+// AllReady composes multiple checkers into one that requires all to pass.
+func AllReady(checkers ...ReadyChecker) ReadyChecker {
```
Comment (jordigilh) [inline internal/handler/health.go:10]: **[P2 QE-1] `AllReady` has no test**

This is an exported public function with branching logic (short-circuit on first failure). Add tests for:
- All checkers pass → returns `true`
- First checker fails → returns `false`
- Last checker fails → returns `false`
- Zero checkers → returns `true`

File: internal/handler/router.go [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+type RouterConfig struct {
+	MetricsRegistry  *metrics.Registry
+	A2AHandler       http.Handler
+	MCPHandler       http.Handler
+	AgentCardHandler http.Handler
+	AuthMiddleware   func(http.Handler) http.Handler
+	ReadyChecker     func() bool
+}
+
+func (c RouterConfig) validate() error {
```
Comment (jordigilh) [inline internal/handler/router.go:20]: **[P2 QE-2] `validate()` has 6 error paths with no individual tests**

Each nil/empty field returns a distinct error message, but no test exercises these paths individually. Add a table-driven test that calls `NewRouter` with each required field set to nil one at a time and asserts the specific error substring.

File: internal/handler/middleware.go [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+
+import (
+	"net/http"
+	"strconv"
+	"time"
+
+	"github.com/jordigilh/kubernaut-apifrontend/internal/metrics"
+)
+
+type statusRecorder struct {
```
Comment (jordigilh) [inline internal/handler/middleware.go:12]: **[P2 ARCH-2] `statusRecorder` doesn't implement `http.Flusher` — will break MCP streaming**

The `metricsMiddleware` wraps the entire mux (line 61 of `router.go`), including `/mcp`. The MCP `StreamableHTTPHandler` requires `http.Flusher` for SSE streaming. When it type-asserts `ResponseWriter` to `http.Flusher`, it will get a `*statusRecorder` which doesn't implement it — streaming will silently fail or panic.

**Fix**: Add a `Flush()` method that delegates to the underlying writer:
```go
func (sr *statusRecorder) Flush() {
    if f, ok := sr.ResponseWriter.(http.Flusher); ok {
        f.Flush()
    }
}
```
Also consider implementing `http.Hijacker` if WebSocket upgrade is ever needed.

File: internal/launcher/launcher.go [hunk 5]
```diff
@@ -0,0 +1,109 @@
...
+// buildAfterExecuteCallback logs task completion with structured context for SRE observability.
+func buildAfterExecuteCallback(log *slog.Logger) adka2a.AfterExecuteCallback {
+	return func(ctx adka2a.ExecutorContext, finalEvent *a2a.TaskStatusUpdateEvent, err error) error {
+		if err != nil {
+			user := auth.UserIdentityFromContext(ctx)
+			username := ""
+			if user != nil {
+				username = user.Username
+			}
+			log.ErrorContext(ctx, "a2a task execution failed",
```
Comment (jordigilh) [inline internal/launcher/launcher.go:126]: **[P2 OPS-1] Error log lacks task ID, session ID, and A2A method**

This error log only includes `"error"` and `"user"`. For SRE triage, also log:
- Task ID from `finalEvent` (the parameter is available but discarded on line 106)
- Session ID if available from context
- The A2A method that failed

Without these, correlating a log entry to a specific user request requires cross-referencing multiple log streams.

File: internal/handler/mcp.go [hunk 6]
```diff
@@ -0,0 +1,101 @@
...
+		}
+		cb := cfg.ToolCallback
+		toolName := t.Name
+		srv.AddTool(toolDef, func(ctx context.Context, _ *mcp.CallToolRequest) (*mcp.CallToolResult, error) {
+			if cb != nil {
+				cb(ctx, toolName)
+			}
+			return &mcp.CallToolResult{
+				Content: []mcp.Content{
+					&mcp.TextContent{Text: fmt.Sprintf("tool %q: not yet wired to backend", toolName)},
```
Comment (jordigilh) [inline internal/handler/mcp.go:93]: **[P2 REL-1] MCP exposes 14 "not yet wired" tool stubs unconditionally**

In a FedRAMP production deployment, `tools/list` will return 14 tools that all respond with `"tool X: not yet wired to backend"`. This is confusing for consumers and expands the attack surface.

Consider:
1. Gating MCP registration behind a feature flag (`ENABLE_MCP=true`)
2. Or not registering tool stubs — register only when the real bridge is ready (PR6+)
3. Or returning a clear MCP error code (not a success with "not implemented" text)

File: internal/launcher/launcher.go [hunk 7]
```diff
@@ -0,0 +1,109 @@
...
+	httpHandler := a2asrv.NewJSONRPCHandler(reqHandler)
+
+	return httpHandler, nil
+}
+
+// buildBeforeExecuteCallback wraps the user-supplied callback so that the
+// UserIdentity from auth middleware is always present in the executor context.
+func buildBeforeExecuteCallback(userCb func(ctx context.Context) (context.Context, error)) adka2a.BeforeExecuteCallback {
+	return func(ctx context.Context, _ *a2asrv.RequestContext) (context.Context, error) {
+		_ = auth.UserIdentityFromContext(ctx)
```
Comment (jordigilh) [inline internal/launcher/launcher.go]: **[P3 SEC-2] Dead code — `UserIdentityFromContext` result discarded**

`_ = auth.UserIdentityFromContext(ctx)` reads the user identity and immediately discards it. This was likely intended to validate that identity is present in context, but since the result is discarded, it provides no actual validation.

Either:
- Remove the dead line
- Or use it to validate: `if user := auth.UserIdentityFromContext(ctx); user == nil { return ctx, fmt.Errorf("missing user identity") }`

File: internal/launcher/launcher.go [hunk 8]
```diff
@@ -0,0 +1,109 @@
...
+			username := ""
+			if user != nil {
+				username = user.Username
+			}
+			log.ErrorContext(ctx, "a2a task execution failed",
+				"error", err,
+				"user", username,
+			)
+		}
+		_ = finalEvent
```
Comment (jordigilh) [inline internal/launcher/launcher.go]: **[P3 SEC-2 addendum] `_ = finalEvent` is also dead code**

This suppresses the unused-variable warning but wastes the `TaskStatusUpdateEvent` that contains the task ID and final state. This data should be logged (per OPS-1) or removed.

File: internal/handler/mcp_test.go [hunk 9]
```diff
@@ -0,0 +1,169 @@
...
+		req = req.WithContext(ctx)
+
+		rec := httptest.NewRecorder()
+		mcpHandler.ServeHTTP(rec, req)
+		Expect(rec.Code).To(Equal(http.StatusOK))
+		Expect(rec.Body.String()).To(ContainSubstring("kubernaut-apifrontend"))
+	})
+
+	It("UT-AF-220-004: tools/list returns registered tools", func() {
+		srv := createMCPServerWithTools()
```
Comment (jordigilh) [inline internal/handler/mcp_test.go]: **[P3 QE-3] Dead code in UT-AF-220-004**

`createMCPServerWithTools()` creates a bare MCP server that is checked for `NotBeNil()` but never used in the actual test assertion. The real test creates a separate `toolHandler` — the `srv` variable adds no value.

Either remove `createMCPServerWithTools()` or replace this test with one that actually exercises `tools/list` over HTTP.

File: internal/launcher/model_test.go [hunk 10]
```diff
@@ -0,0 +1,58 @@
...
+		})
+	})
+
+	Describe("JWT Delegation", func() {
+		It("UT-AF-210-013: JWTDelegationEnabled defaults to true", func() {
+			cfg := launcher.DefaultModelConfig()
+			Expect(cfg.JWTDelegation).To(BeTrue())
+		})
+
+		It("UT-AF-210-014: A2AConfig with model validation", func() {
```
Comment (jordigilh) [inline internal/launcher/model_test.go]: **[P3 QE-4] Tautology test — assigns and reads back a struct field**

```go
cfg := launcher.A2AConfig{AppName: "test"}
Expect(cfg.AppName).To(Equal("test"))
```

This tests Go struct assignment, not any production behavior. Replace with a meaningful assertion (e.g., test that `A2AConfig` with a `ModelConfig` wires correctly) or remove.

File: internal/handler/health.go [hunk 11]
```diff
@@ -0,0 +1,35 @@
...
+}
+
+func handleHealthz(w http.ResponseWriter, _ *http.Request) {
+	w.WriteHeader(http.StatusOK)
+	_, _ = w.Write([]byte("ok"))
+}
+
+func readyzHandler(checker func() bool) http.HandlerFunc {
+	return func(w http.ResponseWriter, _ *http.Request) {
+		if !checker() {
```
Comment (jordigilh) [inline internal/handler/health.go:31]: **[P3 PROD-1] `/readyz` failure message changed**

The previous `main.go` returned `"JWKS circuit breaker open"` on readiness failure. This now returns the generic `"not ready"`. If monitoring scripts or alerts match on the old message string, they'll silently stop matching.

Consider preserving the specific reason or including it in the response body (e.g., `"not ready: JWKS circuit breaker open"`).

File: internal/launcher/model.go [hunk 12]
```diff
@@ -0,0 +1,39 @@
...
+// ModelConfig holds configuration for the LLM model used by the agent.
+type ModelConfig struct {
+	Provider      string
+	Model         string
+	JWTDelegation bool
+}
+
+// DefaultModelConfig returns the default model configuration targeting
+// Claude Sonnet 4.6 via Vertex AI with JWT delegation enabled.
+func DefaultModelConfig() ModelConfig {
```
Comment (jordigilh) [inline internal/launcher/model.go:21]: **[P3 PROD-2] `ModelConfig` and `DefaultModelConfig()` exported but unused in production**

These are defined and tested but never called from `main.go` or any production code path. If planned for a future PR, add a `// TODO(PR6+)` comment linking to the issue. Otherwise, consider deferring these to the PR that actually uses them to avoid dead exports.

File: cmd/apifrontend/main.go [hunk 13]
```diff
@@ -71,60 +77,74 @@ func run() error {
...
+	})
+	if err != nil {
+		return fmt.Errorf("create MCP handler: %w", err)
+	}
+
+	// --- Agent Card ---
+	agentCardHandler, err := handler.NewAgentCardHandler(handler.AgentCardConfig{
+		Name:        "kubernaut-apifrontend",
+		Description: "Kubernaut API Frontend agent for Kubernetes incident triage and remediation",
+		URL:         fmt.Sprintf("https://localhost:%s", envOr("PORT", "8443")),
```
Comment (jordigilh) [inline cmd/apifrontend/main.go]: **[P3 UX-1] Agent Card `url` hardcodes `https://localhost:{PORT}`**

In production, the agent card should reflect the external-facing URL (e.g., from a `PUBLIC_URL` env var or ingress hostname). `localhost` is only valid for local development.

Consider:
```go
URL: envOr("AGENT_CARD_URL", fmt.Sprintf("https://localhost:%s", envOr("PORT", "8443"))),
```

File: internal/handler/health.go (hunk 2 above)
Comment (jordigilh) [inline internal/handler/health.go:10]: **[P3 ARCH-1] `AllReady` exported but never called from production code**

`main.go` passes `validator.Ready` directly as the `ReadyChecker` instead of using `AllReady(validator.Ready)`. This function has no production consumer.

Either:
- Wire it in `main.go`: `ReadyChecker: handler.AllReady(validator.Ready)` (future-proofs for adding more checkers)
- Or keep it unexported until needed

File: internal/launcher/launcher.go [hunk 14]
```diff
@@ -0,0 +1,109 @@
...
+		_ = auth.UserIdentityFromContext(ctx)
+
+		if userCb != nil {
+			return userCb(ctx)
+		}
+		return ctx, nil
+	}
+}
+
+// buildAfterExecuteCallback logs task completion with structured context for SRE observability.
```
Comment (jordigilh) [inline internal/launcher/launcher.go]: **[P2 FED-1 (AU-2/AU-12)] No audit events for A2A task lifecycle or MCP tool invocations**

Neither `BeforeExecuteCallback` nor `AfterExecuteCallback` emit structured audit events via `audit.Emitter`. For FedRAMP AU-2 compliance, the following security-relevant actions need audit trails:
- A2A task created (with user, method, task ID)
- A2A task completed/failed (with user, task ID, duration)
- MCP tool invoked (with user, tool name)

Inject `audit.Emitter` into `A2AConfig` and emit events in both callbacks.

File: cmd/apifrontend/main.go [hunk 15]
```diff
@@ -71,60 +77,74 @@ func run() error {
...
+		Instruction:   "You are the Kubernaut API Frontend agent. Help users triage and remediate Kubernetes incidents.",
+		KABaseURL:     envOr("KA_BASE_URL", "http://localhost:8080"),
+		KAMCPEndpoint: envOr("KA_MCP_ENDPOINT", "http://localhost:8080/api/v1/mcp/"),
+		DSBaseURL:     envOr("DS_BASE_URL", "http://localhost:9090"),
+	}
+	rootAgent, _, err := agentpkg.NewRootAgent(agentCfg)
+	if err != nil {
+		return fmt.Errorf("create root agent: %w", err)
+	}
+	sessionSvc := adksession.InMemoryService()
```
Comment (jordigilh) [inline cmd/apifrontend/main.go:123]: **[P3 REL-3] New env vars undocumented**

This PR introduces several new environment variables (`GCP_PROJECT`, `GCP_REGION`, `KA_BASE_URL`, `KA_MCP_ENDPOINT`, `DS_BASE_URL`, `PORT`) via `envOr()`. Document these in ARCHITECTURE.md or a deployment guide with:
- Purpose
- Default value
- Whether required or optional
- Example values

**Comment by jordigilh** [comment]
**[P2 REL-2] No CHANGELOG or release notes for 5 new endpoints**

This PR introduces major user-facing changes:
- `/a2a/invoke` — A2A JSON-RPC endpoint
- `/mcp` — MCP Streamable HTTP endpoint  
- `/.well-known/agent-card.json` — A2A Agent Card discovery
- `af_http_requests_total` and `af_http_request_duration_seconds` new metrics
- `main.go` rewrite replacing inline routing with `handler.NewRouter`

Add a CHANGELOG entry and deployment notes (new env vars, new routes, monitoring updates).

File: internal/handler/router.go [hunk 16]
```diff
@@ -0,0 +1,62 @@
...
+//   - Authenticated: /a2a/invoke, /mcp
+func NewRouter(cfg RouterConfig) (http.Handler, error) {
+	if err := cfg.validate(); err != nil {
+		return nil, fmt.Errorf("invalid router config: %w", err)
+	}
+
+	mux := http.NewServeMux()
+
+	mux.HandleFunc("GET /healthz", handleHealthz)
+	mux.HandleFunc("GET /readyz", readyzHandler(cfg.ReadyChecker))
```
Comment (jordigilh) [inline internal/handler/router.go:54]: **[P3 OPS-2] `/readyz` only checks JWKS validator readiness**

For a production service with A2A + MCP + session service, readiness should reflect all critical dependencies. The `AllReady` helper in `health.go` exists for exactly this purpose but isn't used.

Consider wiring multiple checkers in `main.go`:
```go
ReadyChecker: handler.AllReady(
    validator.Ready,
    // sessionSvc.Ready (when available),
    // a2aHandler.Ready (when available),
),
```

File: internal/handler/mcp.go [hunk 17]
```diff
@@ -0,0 +1,101 @@
...
+			}, nil
+		})
+	}
+
+	handler := mcp.NewStreamableHTTPHandler(
+		func(_ *http.Request) *mcp.Server { return srv },
+		nil,
+	)
+
+	return handler, nil
```
Comment (jordigilh) [inline internal/handler/mcp.go:104]: **[P3 FED-2 (SI-10)] MCP tool stubs use empty input schemas**

```go
objectSchema := map[string]any{
    "type":       "object",
    "properties": map[string]any{},
}
```

All 14 tools accept any JSON input without validation. When the real tool bridge is wired (PR6+), each tool must have a proper JSON Schema defining required fields, types, and constraints per NIST 800-53 SI-10 (Information Input Validation).

Consider adding a `// TODO(PR6+): replace with generated schemas from ADK tool structs` comment.

File: internal/handler/middleware.go (hunk 1 above)
Comment (jordigilh) [reply internal/handler/middleware.go]: Fixed. `normalizePath()` now maps URL paths to fixed labels before using them as Prometheus label values. See `internal/handler/middleware.go`.

File: internal/handler/health.go (hunk 2 above)
Comment (jordigilh) [reply internal/handler/health.go:10]: Fixed. Added tests for `AllReady` covering short-circuit on first failure, all-pass, and single-checker cases. See `internal/handler/handler_test.go`.

File: internal/handler/router.go (hunk 3 above)
Comment (jordigilh) [reply internal/handler/router.go:20]: Fixed. Added individual tests for all 6 `RouterConfig.validate()` error paths using `DescribeTable`. See `internal/handler/handler_test.go`.

File: internal/handler/middleware.go (hunk 4 above)
Comment (jordigilh) [reply internal/handler/middleware.go:12]: Fixed. `statusRecorder` now implements `http.Flusher` via a `Flush()` method that delegates to the underlying `ResponseWriter` if it supports flushing. MCP streaming works correctly.

File: internal/launcher/launcher.go (hunk 5 above)
Comment (jordigilh) [reply internal/launcher/launcher.go:126]: Fixed. `AfterExecuteCallback` now logs `task_id`, `session_id` (from ExecutorContext), and `method`. Also emits `EventA2ATaskCompleted`/`EventA2ATaskFailed` audit events.

File: internal/handler/mcp.go (hunk 6 above)
Comment (jordigilh) [reply internal/handler/mcp.go:93]: Fixed. MCP handler is now gated behind `mcp.enabled` config field. When disabled, `/mcp` returns 501 'MCP not enabled'. Tests updated to pass `Enabled: true` explicitly.

File: internal/launcher/launcher.go (hunk 7 above)
Comment (jordigilh) [reply internal/launcher/launcher.go]: Fixed. Removed dead `_ = auth.UserIdentityFromContext(ctx)` — the user identity is now passed to the audit emitter directly in the callback.

File: internal/launcher/launcher.go (hunk 8 above)
Comment (jordigilh) [reply internal/launcher/launcher.go]: Fixed. Removed `_ = finalEvent` dead code. The `finalEvent` is now used to extract `TaskID` for the audit log.

File: internal/handler/mcp_test.go (hunk 9 above)
Comment (jordigilh) [reply internal/handler/mcp_test.go]: Fixed. Removed the unused `createMCPServerWithTools()` helper and its dead import.

File: internal/launcher/model_test.go (hunk 10 above)
Comment (jordigilh) [reply internal/launcher/model_test.go]: Fixed. Replaced tautology test with a meaningful assertion: `NewA2AHandler` returns an error when `Agent` is nil in `A2AConfig`.

File: internal/handler/health.go (hunk 11 above)
Comment (jordigilh) [reply internal/handler/health.go:31]: Acknowledged. Added `// TODO(PR7+): consider ReadyChecker returning (bool, string) for specific reasons.` comment to the readyz handler.

File: internal/launcher/model.go (hunk 12 above)
Comment (jordigilh) [reply internal/launcher/model.go:21]: Acknowledged. Added `// TODO(PR6+): wired when model selection is configurable via env/CRD.` comment to `DefaultModelConfig()`.

File: cmd/apifrontend/main.go (hunk 13 above)
Comment (jordigilh) [reply cmd/apifrontend/main.go]: Fixed. Agent Card URL is now loaded from `agentCard.url` in the ConfigMap config file. If omitted, it is derived from `server.port` via `ResolveDefaults()`. No more hardcoded localhost.

File: internal/handler/health.go (hunk 2 above)
Comment (jordigilh) [reply internal/handler/health.go:10]: Fixed. `main.go` now uses `handler.AllReady(validator.Ready)` — wiring it into production code. Added `// TODO(PR7+)` for session/A2A checkers.

File: internal/launcher/launcher.go (hunk 14 above)
Comment (jordigilh) [reply internal/launcher/launcher.go]: Fixed. Added audit events: `EventA2ATaskStarted` in BeforeExecuteCallback, `EventA2ATaskCompleted`/`EventA2ATaskFailed` in AfterExecuteCallback, and `EventMCPToolInvoked` in the MCP tool handler. Auditor wired via config in both handlers.

File: cmd/apifrontend/main.go (hunk 15 above)
Comment (jordigilh) [reply cmd/apifrontend/main.go:123]: Fixed. Env vars have been completely removed. Configuration is now loaded from a YAML file (ConfigMap mount) via `--config` flag. All fields documented in `docs/design/ARCHITECTURE.md` under the new 'Configuration' section with a full field table.

File: internal/handler/router.go (hunk 16 above)
Comment (jordigilh) [reply internal/handler/router.go:54]: Addressed. `main.go` now uses `handler.AllReady(validator.Ready)` for the readiness checker. Added `// TODO(PR7+): add session/A2A readiness checkers` for when those services have Ready() methods.

File: internal/handler/mcp.go (hunk 17 above)
Comment (jordigilh) [reply internal/handler/mcp.go:104]: Acknowledged. Added `// TODO(PR6+): replace with generated schemas from ADK tool structs per SI-10.` comment to `DefaultMCPTools()`. Real schemas will be wired when the tool bridge is implemented.

**Comment by jordigilh** [comment]
**REL-2 (CHANGELOG):** Added `CHANGELOG.md` documenting all 6 new endpoints, metrics middleware, audit events, ConfigMap-based configuration, and the breaking change (envOr removal) with a migration guide.

File: cmd/apifrontend/main.go (hunk 13 above)
Comment (jordigilh) [reply cmd/apifrontend/main.go]: Fixed. Agent Card URL is now loaded from `agentCard.url` in the ConfigMap config file. If omitted, it is derived from `server.port` via `ResolveDefaults()`. No more hardcoded localhost.

File: internal/handler/health.go (hunk 2 above)
Comment (jordigilh) [reply internal/handler/health.go:10]: Fixed. `main.go` now uses `handler.AllReady(validator.Ready)` — wiring it into production code. Added `// TODO(PR7+)` for session/A2A checkers.

File: internal/launcher/launcher.go (hunk 14 above)
Comment (jordigilh) [reply internal/launcher/launcher.go]: Fixed. Added audit events: `EventA2ATaskStarted` in BeforeExecuteCallback, `EventA2ATaskCompleted`/`EventA2ATaskFailed` in AfterExecuteCallback, and `EventMCPToolInvoked` in the MCP tool handler. Auditor wired via config in both handlers.

File: cmd/apifrontend/main.go (hunk 15 above)
Comment (jordigilh) [reply cmd/apifrontend/main.go:123]: Fixed. Env vars have been completely removed. Configuration is now loaded from a YAML file (ConfigMap mount) via `--config` flag. All fields documented in `docs/design/ARCHITECTURE.md` under the new 'Configuration' section with a full field table, YAML structure reference, and ConfigMap mount example.

File: internal/handler/router.go (hunk 16 above)
Comment (jordigilh) [reply internal/handler/router.go:54]: Addressed. `main.go` now uses `handler.AllReady(validator.Ready)` for the readiness checker. Added `// TODO(PR7+): add session/A2A readiness checkers` for when those services have Ready() methods.

File: internal/handler/mcp.go (hunk 17 above)
Comment (jordigilh) [reply internal/handler/mcp.go:104]: Acknowledged. Added `// TODO(PR6+): replace with generated schemas from ADK tool structs per SI-10.` comment to `DefaultMCPTools()`. Real schemas will be wired when the tool bridge is implemented.
